upgrade_test.py

```diff
@@ -23,6 +22,36 @@
 from sdcm.nemesis import RollbackNemesis
 from sdcm.nemesis import UpgradeNemesis
 from sdcm.fill_db_data import FillDatabaseData
+from functools import wraps
+
+
+def truncate_entries(func):
+    @wraps(func)
+    def _inner(self, *args, **kwargs):
+        # Perform validation of truncate entries in case the new version is 3.1 or more
+        node = args[0]
+        new_version = self.params.get('new_version', default='')
+        new_ver = '.'.join(n for n in new_version.split('.')[:2])
+        if new_ver and float(new_ver) >= 3.1:
+            base_version = self.params.get('scylla_version', default='')
+            system_truncated = True if base_version.startswith('3.1') else False
+
+            with self.cql_connection_patient(node) as session:
+                self.cql_truncate_table_for_truncate_test(session=session)
+                self.validate_truncated_entries_for_table(session=session, system_truncated=system_truncated)
+
+        func(self, *args, **kwargs)
+
+        result = node.remoter.run('scylla --version')
+        new_ver = result.stdout
+        new_ver = '.'.join(n for n in new_ver.split('.')[:2])
+        if new_ver and float(new_ver) >= 3.1:
```

Comment:
I would recommand using `from distutils.version import LooseVersion` for comapring versions, assuming version structure like that or that it would be a float, might fail later on.
https://android.googlesource.com/toolchain/python/+/0d80e55d38276ed373a55fa4a8f69d62ef5d7408/Python-2.7.3/Lib/distutils/version.py#228

Reply:
`from pkg_resources import parse_version` is also a good option.

Reply:
LooseVersion works wrong (return wrong result). parse_version is better. Thanks